test/runtests.jl

```diff
@@ -118,6 +118,16 @@ ch2 = chain(1:0, 1:2:5, 0.2:0.1:1.6)
 @test eltype(ch2) == typejoin(Int, Float64)
 @test collect(ch2) == [1:2:5; 0.2:0.1:1.6]
 
+ch3 = chain(1:10,1:10, 1:10)
+@test length(ch3) == 30
+@test Base.iteratorsize(ch3) == Base.HasLength()
```

Comment:
ah, I see, sorry, I was looking at 0.5 travis build

I can switch Base.Iteratorsize to Interface.Iteratorsize but we still have a problem with Base.HasLength(). Should I put the whole test inside a
```julia
if VERSION >= v"0.5.0-dev+3305"
...
end
```
?